### PR TITLE
Implement Gnocchi resource update

### PR DESCRIPTION
**Pull request by ozerovandrei**
Add Gnocchi resource create request with tests and documentation.

The same command with Gnocchi CLI is done with:
`gnocchi resource update`

For [#698](https://github.com/gophercloud/gophercloud/issues/698)

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://github.com/gnocchixyz/gnocchi/blob/stable/4.1/gnocchi/rest/api.py#L1007
https://github.com/gnocchixyz/gnocchi/blob/stable/4.1/gnocchi/indexer/sqlalchemy.py#L788

### CODE REVIEW

**Comment by ozerovandrei** [comment]
@jtopjian this PR has been rebased with master. And I added the `required` tag for Resource `CreateOpts` `ID` field as dicsussed in #16.

File: gnocchi/metric/v1/resources/requests.go [hunk 1]
```diff
@@ -132,3 +132,48 @@ func Create(client *gophercloud.ServiceClient, resourceType string, opts CreateO
...
+type UpdateOptsBuilder interface {
+	ToResourceUpdateMap() (map[string]interface{}, error)
+}
+
+// UpdateOpts represents options used to update a network.
+type UpdateOpts struct {
+	// Metrics field can be used to link existing metrics in the resource
+	// or to create metrics and update the resource at the same time to save
+	// some requests.
+	Metrics map[string]interface{} `json:"metrics,omitempty"`
```
Comment (jtopjian) [inline gnocchi/metric/v1/resources/requests.go]: In order to delete a metric from a resource, Gnocchi wants an empty `metrics` object:

```
REQ: curl -g -i -X PATCH http://127.0.0.1:8041/v1/resource/generic/fba056ff-60f3-4edc-bb17-7c84531fbb53 -H "User-Agent: gnocchi keystoneauth1/3.1.0 python-requests/2.14.2 CPython/2.7.5" -H "Content-Type: application/json" -H "Accept: application/json, */*" -H "X-Auth-Toke
n: {SHA1}122e9eefd8a4fc1ead3e2cbce2b564b7c147d175" -d '{"metrics":{}}'
```

In order to do this, we need to change the above to:

```go
Metrics *map[string]interface{} `json:"metrics,omitempty"`
```

This way metrics can be deleted by specifying a pointer to a zero-value map. In addition, not specifying a `Metrics` field will not inadvertently cause metrics to be deleted.

File: gnocchi/metric/v1/resources/requests.go [hunk 2]
```diff
@@ -132,3 +132,48 @@ func Create(client *gophercloud.ServiceClient, resourceType string, opts CreateO
...
+
+	// ProjectID is the Identity project of the resource.
+	ProjectID string `json:"project_id,omitempty"`
+
+	// UserID is the Identity user of the resource.
+	UserID string `json:"user_id,omitempty"`
+
+	// StartedAt is a resource creation timestamp.
+	StartedAt string `json:"started_at,omitempty"`
```
Comment (jtopjian) [inline gnocchi/metric/v1/resources/requests.go]: This needs to be `time.Time`, similar to `CreateOpts`

File: gnocchi/metric/v1/resources/requests.go [hunk 3]
```diff
@@ -132,3 +132,48 @@ func Create(client *gophercloud.ServiceClient, resourceType string, opts CreateO
...
+
+	// EndedAt is a timestamp of when the resource has ended.
+	EndedAt string `json:"ended_at,omitempty"`
```
Comment (jtopjian) [inline gnocchi/metric/v1/resources/requests.go]: This needs to be `time.Time`, similar to `CreateOpts`.

**Comment by ozerovandrei** [comment]
@jtopjian thank you for a review. I've fixed all that stuff and rebased it from the master.

**Pull request merged**